oboformat/src/main/java/org/obolibrary/obo2owl/OWLAPIOwl2Obo.java

```diff
@@ -1761,10 +1769,29 @@ public static String getIdentifierFromObject(OWLObject obj, @Nonnull OWLOntology
      */
     @Nullable
     public static String getIdentifier(@Nullable IRI iriId) {
+        return getIdentifier(iriId, new TreeMap<>());
+    }
+
+    /**
+     * See table 5.9.2. Translation of identifiers
+     *
+     * @param iriId the iri id
+     * @param namespaceToPrefixMap map to pass through for IRI compaction
+     * @return obo identifier or null
+     */
+    @Nullable
+    public static String getIdentifier(@Nullable IRI iriId, @Nonnull SortedMap<String, String> namespaceToPrefixMap) {
         if (iriId == null) {
             return null;
         }
         String iri = iriId.toString();
+        boolean match = namespaceToPrefixMap.entrySet().stream().anyMatch(e -> iri.startsWith(e.getKey()));
+        Optional<Map.Entry<String, String>> mappingOpt = namespaceToPrefixMap.entrySet().stream().filter(e -> iri.startsWith(e.getKey())).findFirst();
```

Comment:
A lot of places in this PR are more re-implementstioms of curie and iri conversion. Maybe we can directly start using curies4j here? 

All I need to do still is figure out how to get it in maven. Alternatively, we can vendor it fully in owlapi

Reply:
I think we could use it once it's in maven, assuming OWLAPI is okay with adding a dependency. We can switch to that at any point; doesn't have to be in this PR. Actually there are probably many other curie/iri conversions in OWLAPI. I probably ought to see if there is something in here I should be reusing.

Reply:
great, thanks for the perspective. I think that makes sense.

Reply:
Thanks for the nudge @cthoyt; I realized I should just be using the OWL API PrefixManager directly, which I already had a reference to.